### DISCUSSION TITLE
Unable to change color of datagrid header and rows separator

### DISCUSSION THREAD

**Discussion by Varmod** [AvaloniaUI/Avalonia#12399] [Q&A] [answered]
**Describe the bug**
I'm unable to change the color of the datagrid header and rows separator. After I apply a style to it nothing happens.
In devtools the style properly shows up and is active, however the colors stays default #66000000. I'm also unable to change the color in devtools manually.
Same thing applies to the datagrids TopLeft Header where I'm unable to change both the vertical and horizontal separator.

**To Reproduce**
apply a style as following:
`<Style Selector="DataGrid /template/ Rectangle#PART_ColumnHeadersAndRowsSeparator">
    <Setter Property="Fill" Value="Navy" />
</Style>`
see nothing change.

**Desktop**
- OS: Windows
- Version 11.0.1

**Additional info**
The controls should probably be a default target for the horizontal and vertical grid lines property and shouldn't be needed to be edited extra.

![image](https://github.com/AvaloniaUI/Avalonia/assets/52281562/84cd4287-6302-47a3-9aeb-8f4ac69ff619)

**Comment by timunie** [comment, marked as answer]
you cannot override locally set values in a Style: 
https://github.com/AvaloniaUI/Avalonia/blob/6a80ef86f357fbf495bbb7b0ebd3dd94c0732d21/src/Avalonia.Controls.DataGrid/Themes/Fluent.xaml#L528C49-L528C71

However, you can add override the dynamic resource in App.Resources:
```xml
<SolidColorBrush x:Key="DataGridGridLinesBrush">Navi</SolidColorBrush>
```

Note: I'll convert this into a Q&A as I think it's by design.

**Reply by Varmod** [reply]
Thanks, this does fix it for me. 
But this approach doesn't seem inline with how we do it with styles. It's an extra place I have to edit to make a change I already declared somewhere else.
